src/Native/System.Native/pal_mount.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * Struct to describe the amount of free space and total space on a given mount point
+ */
+struct MountPointInformation
+{
+    int64_t AvailableFreeSpace;
+    int64_t TotalFreeSpace;
+    int64_t TotalSize;
+};
+
+/**
+ * Function pointer to call back into C# when we find a mount point via GetAllMountPoints. 
+ * Using the callback pattern allows us to limit the number of allocs we do and makes it 
+ * cleaner on the managed side since we don't have to worry about cleaning up any unmanaged memory.
+ */
+typedef void (*MountPointFound)(char* name, int32_t nameLength);
```

Comment:
Ah good idea adding const-correctness. Making const char * const for complete correctness